frontend/src/pages/roles/components/CreateNewRole.tsx

```diff
@@ -0,0 +1,112 @@
+import Switch from '@mui/material/Switch';
+import { useEffect, useRef, useState } from 'react';
+import tinycolor from 'tinycolor2';
+import RoleDTO from '@shared/dtos/RoleDTO';
+import { Permission } from '@shared/utils/Permission';
+import { useUserContext } from '../../../contexts/UserContext';
+import Button from '../../../utility_components/Button';
+
+const PermissionsLists: { roles: Permission[]; users: Permission[]; meetings: Permission[] } = {
+    roles: ['VIEW_ROLES', 'MANAGE_ROLES'],
+    users: ['VIEW_USERS', 'MANAGE_USERS'],
+    meetings: ['VIEW_MEETINGS', 'MANAGE_MEETINGS'],
+};
+
+export function CreateNewRole({
+    roles,
+    setRoles,
+}: {
+    roles: Record<string, RoleDTO>;
+    setRoles: (val: Record<string, RoleDTO>) => void;
+}) {
+    const [submitDisabled, setSubmitDisabled] = useState<boolean>(false);
+    const [permissions, setPermissions] = useState<Set<Permission>>(new Set());
+    const nameRef = useRef<HTMLInputElement>(null);
+    const userContext = useUserContext();
+
+    function checkSwitch(perm: Permission) {
+        if (permissions.has(perm)) {
+            permissions.delete(perm);
+        } else {
+            permissions.add(perm);
+        }
+        setPermissions(new Set(Array.from(permissions)));
+    }
+
+    async function createNewRole() {
+        if (nameRef.current === null) {
+            return;
+        }
+
+        if (nameRef.current.value.trim() === '') {
+            return;
+        }
+
+        setSubmitDisabled(true);
+        const data = await userContext.fetcher('POST /api/roles', {
+            color: tinycolor.random().toHexString(),
+            name: nameRef.current.value,
+            permissions: Array.from(permissions),
+        });
+
+        setSubmitDisabled(false);
+
+        if (data) {
+            setRoles({ ...roles, [data.role.id]: data.role });
+            setPermissions(new Set());
+            nameRef.current.value = '';
+        }
+    }
+
+    useEffect(() => {
```

Comment:
Let's remove this console.log

Reply:
lmfao just realised that you can't see the clg, its just the line below it

Reply:
Done